lib/state-controller.js

```diff
@@ -76,66 +69,60 @@ const StateController = {
   downloadKite(url, opts) {
     opts = opts || {};
     return this.canInstallKite()
-    .then(() => utils.spawnPromise('curl', [
-      '-L', url, '--output', this.KITE_DMG_PATH,
-    ], 'curl_error'))
+    .then(() => this.streamKiteDownload(opts.onDownloadProgress))
     .then(() => utils.guardCall(opts.onDownload))
     .then(() => opts.install && this.installKite(opts));
   },
 
-  installKite(opts) {
-    opts = opts || {};
+  streamKiteDownload(progress) {
```

Comment:
Yes, I remember, but I really think it was some misconfiguration or misuse of the stream used at that point. Once again, stream were one of the big selling point of node and are stable since a long time. Last friday I asked @alexflint to make sure this version works for you, oviously I'll revert if we see the same issue.

The rationale for the change is that relying on an external app to download the file will make supporting different platform tedious. And we can't get any information about the download process that way.

To test this change quickly you can do as follow:
- Open Atom in the kite-installer directory while in this branch.
- Open the devtools and run the following code:
```js
// get the path to the state controller
p = atom.project.getPaths()[0]
scp = p + '/lib/state-controller'; 

StateController = require(scp)

// starts the download and display its progress
StateController.streamKiteDownload((b,t,r) => console.log(Math.round(r * 100) + "%")).then(() => console.log('done'))